libs/community/langchain_community/chat_message_histories/sql.py

```diff
@@ -80,36 +92,85 @@ def get_sql_model_class(self) -> Any:
         return self.model_class
 
 
+DBConnection = Union[AsyncEngine, Engine, str]
+
+
 class SQLChatMessageHistory(BaseChatMessageHistory):
     """Chat message history stored in an SQL database."""
 
+    @property
+    def Session(self) -> Union[scoped_session, async_sessionmaker]:
+        logger.warning("Session is deprecated, use session_maker instead")
+        return self.session_maker
+
     def __init__(
         self,
         session_id: str,
-        connection_string: str,
+        connection_string: Optional[str] = None,
+        connection: Union[None, DBConnection] = None,
+        engine_args: Optional[Dict[str, Any]] = None,
+        async_mode: Optional[bool] = None,  # Use only if connection is a string
         table_name: str = "message_store",
         session_id_field_name: str = "session_id",
         custom_message_converter: Optional[BaseMessageConverter] = None,
     ):
-        self.connection_string = connection_string
-        self.engine = create_engine(connection_string, echo=False)
+        if connection_string:
+            logger.warning("connection_string is deprecated, use connection instead")
```

Comment:
Could you swap to use `warn_deprecated` (git grep for usage of `warn_deprecated`)

You may need to memoize it so it doesn't flood the user with warnings on repeated instantiations.